src/doc/trpl/dining-philosophers.md

```diff
@@ -0,0 +1,973 @@
+        Philosopher {
+            name: name.to_string(),
+            done: done,
+        }
+    }
+```
+
+Now that we have a second field for our struct, we need to define it in
+`new()`. In this case, we’re going to take it as an argument. We’ll pass this
+in in `main()`, below.
+    
+```rust,ignore
+    fn done(&self) {
+        println!("{} is done eating.", self.name);
+
+        self.done.send(true).ok().expect("Couldn't finish eating");
+    }
+}
+```
+
+We’ve added a new line to `done()`. We use the `send()` method on `Sender`
+to send `true` down the channel. `send()` returns a `Result`, letting us
+know if we’ve had an error[^error1] while trying to send the value. In this
+case, we want to just crash and throw an error if it doesn’t work, so we use
+the `ok().expect()` pattern we used in the Guessing Game.
+
+[error1]: There’s one possible error here: the receiving end has disconnected.
+
+```rust,ignore
+fn main() {
+    let (done_tx, done_rx) = mpsc::channel();
+```
+
+We need to actually create a channel, and the `mpsc::channel()` function does
+that. It returns both ends of the channel, the sending and receiving end. We
+can use a single `let` to bind both ends at the same time, setting both
+`done_tx` and `done_rx`. The `tx` and `rx` stand for ‘transmission’ and
+‘receiver’.
+
+```rust,ignore
+    let philosophers = vec![
+        Philosopher::new("Baruch Spinoza", done_tx.clone()),
+        Philosopher::new("Gilles Deleuze", done_tx.clone()),
+        Philosopher::new("Karl Marx", done_tx.clone()),
+        Philosopher::new("Friedrich Nietzsche", done_tx.clone()),
+        Philosopher::new("Michel Foucault", done_tx.clone()),
+    ];
+```
+
+Now that we’ve gotten a channel, we want each philosopher to have a copy of the
+sending end. The `clone()` method on `Sender` lets us do that. Now we have a
+channel with the receiving end in `main()`, and the sending end in each one of
+our `Philosopher`s.
+
+```rust,ignore
+    for _ in 0..5 {
+        done_rx.recv().ok().expect("The channel was closed prematurely.");
+    }
+}
+```
+
+The `recv()` method is on the receiving end of the channel, which has the type
+`Receiver`. Since we have five philosophers, we loop five times, and ask for
+five values on the channel. This all matches up, and our program runs. You should
+get the same output as before: all five philosophers reporting in.
+
+Next, we want to make our philosophers not just finish eating, but actually
+eat. Here’s the next version:
+
+```rust
+use std::sync::mpsc;
+use std::sync::mpsc::Sender;
+use std::thread;
+
+struct Philosopher {
+    name: String,
+    done: Sender<bool>
+}   
+
+impl Philosopher { 
+    fn new(name: &str, done: Sender<bool>) -> Philosopher {
+        Philosopher {
+            name: name.to_string(),
+            done: done,
+        }
+    }
+    
+    fn done(&self) {
+        println!("{} is done eating.", self.name);
+
+        self.done.send(true).ok().expect("Couldn't finish eating");
```

Comment:
I'm still kinda worried about all these `.ok().expect(...)` calls throughout these examples, some of them I think may be giving the wrong impression. For example this doesn't really indicate anything about "couldn't finish eating" but rather that no one's listening for the notification that the eating is done. Down below there's:

```rust
.ok().expect("Couldn't aquire left mutex");
```

If `Err` is returned, though, the mutex was indeed acquired, it was just poisoned. 

Would it be possible to just use `unwrap` everywhere? I would consider `unwrap` as the idiomatic way to indicate that an error should not ever be happening here, and errors should definitely not ever be happening in these examples.